src/contexts/AuthContext.tsx

```diff
@@ -11,20 +11,42 @@ export type TRole = Role | null | undefined;
 export interface AuthContext {
   isAuthenticated: boolean;
   isAdmin: boolean;
+  login: (client: SpvWalletClientExtended) => void;
+  logout: () => void;
 }
 
 const AuthContext = createContext<AuthContext | null>(null);
 
 export const AuthProvider = ({ children }: { children: React.ReactNode }) => {
-  const { spvWalletClient } = useContext(SpvWalletContext) || {};
-
-  const isAuthenticated = !!spvWalletClient;
-  const isAdmin = isAuthenticated && spvWalletClient?.role === Role.Admin;
-  const contextValue = {
-    isAuthenticated,
-    isAdmin,
-  };
-  return <AuthContext.Provider value={contextValue}>{children}</AuthContext.Provider>;
```

Comment:
Why have you changed this provider? It seemed good to me.

Reply:
It was an issue with redirecting after login

Reply:
If so, I'm not really convinced about your solution. But for now, if it works, don't change it. 
But if you could, just rethink your implementation and maybe you'll figure out how to do it differently.

Reply:
Simplified logic